### DISCUSSION TITLE
Lesson 7

### DISCUSSION THREAD

**Discussion by nayashi002** [Cyfrin/foundry-full-course-cu#1856] [General]
i tried writing a test to check if the address has been reset to zero after withdrawing, using the same logic as patrick previous test of checking the Multiple users funding the contract, after testing the particular tests it passes but my coverage for the fundMe.sol does not increase instead it remains the same can someone help am currently in lecture 7 and below is the test

``function testEmptyAddressAfterWithdrawal() public funded{
   uint160 startingFunders = 1;
   uint160 numberOfFunders = 10;
   for(uint160 i = startingFunders; i < numberOfFunders; i++){
        
        hoax(address(i),STARTING_BALANCE);
        fundMe.fund{value:STARTING_BALANCE}();
   }
   vm.startPrank(fundMe.getOwner());
   fundMe.withdraw();
   vm.stopPrank();
   numberOfFunders = uint160(address(0));
   assert(numberOfFunders == 0);

}``

**Comment by urosognjenovic** [comment]
The coverage will not increase if you're testing the functions/lines/branches that have been tested in your previous tests.

In this case, you're calling the functions ```fund``` and ```withdraw```. Please check if you've tested these in the other tests.

**Reply by nayashi002** [reply]
It seems my code was wrong am trying to implement that after withdrawing, the array of s_funder which we use a getter of  getfunders() get reset back to an empty array my code logic is wrong.. am just trying to boost the coverage of my FundMe...I now get that an implementing an already use codeblock..thanks will still try figuring it out

**Reply by urosognjenovic** [reply]
I noticed that your implementation is wrong, but I didn't want to get off topic. What you need to do is access the length of the ```s_funders``` array. You can create an external view function for this:
```
function getFundersLength() external view returns (uint256) {
    return s_funders.length;
}
```
Then, in your test add the following code:
```
uint256 fundersLength = fundMe.getFundersLength();
assertEq(fundersLength, 0);
```
On topic, I tested Patrick's code to confirm my initial assumption. Removing the function that tests the withdraw function from multiple funders doesn't affect the coverage for the reasons mentioned above.

**Reply by nayashi002** [reply]
Wow thanks for this code i will try it out to see...
by the way would love to connect with you are you on whatsapp ?

**Reply by nayashi002** [reply]
Thanks for the code actually improved the coverage by a little percentage i was just curious on writing test to improve it to 100%..
Still expecting to connect with you.
